### PR TITLE
json-writer: replace invalid sequences in messages with �

### PR DESCRIPTION
**Pull request by lzaoral**
Fixes: #42

### CODE REVIEW

**Review by kdudka — CHANGES_REQUESTED** [review]
I like the solution but it needs to be extended to `sarifEncodeMsg()`, too:
```sh
$ echo $'\xb4' > x.c && echo 'x.c:1: warning: msg' | csgrep --mode=sarif -U1 | csgrep
-:71: error: invalid code sequence
```

File: src/json-writer.cc [hunk 1]
```diff
@@ -28,8 +28,18 @@
...
 #include <boost/iostreams/filter/regex.hpp>
+#include <boost/nowide/utf/convert.hpp>
 #include <boost/property_tree/json_parser.hpp>
 
+static inline std::string sanitizeUTF8(const std::string &str)
+{
+    using boost::nowide::utf::convert_string;
+
+    // every non-UTF8 sequence will be replaced with �
+    // � - REPLACEMENT CHARACTER U+FFFD
```
Comment (kdudka) [inline src/json-writer.cc:39]: I would prefer not to have the special character directly in the source code.  Could we just write `0xEF 0xBF 0xBD` in the comment?

**Comment by kdudka** [comment]
I am marking this as draft until we resolve the compatibility with EPEL buildroots.

File: src/json-writer.cc (hunk 1 above)
Comment (lzaoral) [reply src/json-writer.cc]: Addressed in 620dc470d46908c64d3a5d14c78e015c71836aee.

**Comment by kdudka** [comment]
Thanks!  I hit this difference in behavior while I was working on PR #48 yesterday.

**Comment by lzaoral** [comment]
Thanks for the review, @kdudka! Merging.

**Comment by lzaoral** [comment]
Thanks @praiskup! Addressed in f7e5c14fbbd0c0f48afdba50e4e7d427da48fce0. I'll merge this PR when it's green.

**Pull request merged**